### PR TITLE
[INTEL MKL] [2.5 cherry pick]  Log warning message for int8 in case of native format

### PR DESCRIPTION
**Pull request by gzmkl**
This PR logs a warning message to the Tensorflow customers in the following cases when OneDNN optimization does not 
support INT8 inference/training:

   (1) stock Tensorflow, with OneDNN optimizations enabled (in this case, INT8 is not supported at all.
 
   (2) Intel Optimized Tensorflow (aka, build with --config=mkl): in this case, the warning message reminds the user
        to set a proper environment variable before running INT8 inference/training. 

Please help to cherry pick this small PR into RC2 branch.

### CODE REVIEW

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 1]
```diff
@@ -3688,6 +3689,24 @@ MklLayoutRewritePass::CheckForQuantizedNodeRewrite(const Node* n) const {
...
+          static absl::once_flag once;
+	  absl::call_once(once, [] {
+#if defined(ENABLE_MKL)
+            VLOG(0) << "MklLayoutRewritePass::RewriteInfo: Do not support INT8"
+                    << "data type for native format. Please set environment "
+                    << "variable TF_ENABLE_MKL_NATIVE_FORMAT to false.";
+#else
+            VLOG(0) << "MklLayoutRewritePass::RewriteInfo: Do not support INT8 "
+	            << "data type for native format. Please switch to Intel "
+	            << "Optimized Tensorflow and set environment variable "
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:3704]: Nit: Could you please fix the indentation? Thank you!
```suggestion
            VLOG(0) << "MklLayoutRewritePass::RewriteInfo: Do not support INT8 "
	              << "data type for native format. Please switch to Intel "
	              << "Optimized Tensorflow and set environment variable "
```

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 2]
```diff
@@ -3688,6 +3689,24 @@ MklLayoutRewritePass::CheckForQuantizedNodeRewrite(const Node* n) const {
   if (type_attrs_present) {
     for (auto ri = rinfo_.cbegin(); ri != rinfo_.cend(); ++ri) {
       if (n->type_string().compare(ri->name) == 0 && ri->rewrite_rule(n)) {
+	// Currently OneDNN optimization does not support int8 with native
+	// format.
+        if (NativeFormatEnabled()) {
+          static absl::once_flag once;
+	  absl::call_once(once, [] {
+#if defined(ENABLE_MKL)
+            VLOG(0) << "MklLayoutRewritePass::RewriteInfo: Do not support INT8"
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:3698]: Nit: This makes it clearer that the subject is `MklLayoutRewritePass::RewriteInfo`. Same for the other message.
```suggestion
            VLOG(0) << "MklLayoutRewritePass::RewriteInfo does not support INT8"
```

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 3]
```diff
@@ -3688,6 +3689,24 @@ MklLayoutRewritePass::CheckForQuantizedNodeRewrite(const Node* n) const {
...
+                    << "data type for native format. Please set environment "
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:3699]: Nit:
```suggestion
                    << "data type for native format. Please set the environment "
```
Same for the other message.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

**Comment by gzmkl** [comment]
@penpornk Thanks for review suggestions. Code change has been committed.

File: tensorflow/core/common_runtime/mkl_layout_pass.cc (hunk 3 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/mkl_layout_pass.cc]: done

File: tensorflow/core/common_runtime/mkl_layout_pass.cc (hunk 2 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/mkl_layout_pass.cc]: changed

File: tensorflow/core/common_runtime/mkl_layout_pass.cc (hunk 1 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/mkl_layout_pass.cc]: fixed

**Review by penpornk — APPROVED** [review]
Thank you for the fix!

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 4]
```diff
@@ -3688,6 +3689,24 @@ MklLayoutRewritePass::CheckForQuantizedNodeRewrite(const Node* n) const {
...
+          static absl::once_flag once;
+          absl::call_once(once, [] {
+#if defined(ENABLE_MKL)
+            VLOG(0) << "MklLayoutRewritePass::RewriteInfo does not support INT8"
+                    << "data type for native format. Please set the environment"
+                    << " variable TF_ENABLE_MKL_NATIVE_FORMAT to false. ";
+#else
+            VLOG(0) << "MklLayoutRewritePass::RewriteInfo does not support INT8"
+                    << " data type for native format. Please switch to Intel "
+                    << "Optimized Tensorflow and set environment variable "
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:3704]: Nit: Could you please fix this here too?
```suggestion
                    << "Optimized Tensorflow and set the environment variable "
```

**Review by penpornk — CHANGES_REQUESTED** [review]

File: tensorflow/core/common_runtime/mkl_layout_pass.cc (hunk 4 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/mkl_layout_pass.cc]: @penpornk  

Done. I should have checked the 2nd message when fixing the 1st one :)

**Review by penpornk — APPROVED** [review]
Thank you again! :)

**Comment by gzmkl** [comment]
@penpornk We would like to have this PR cherry-picked into TF 2.5-rc2. Thanks!

**Comment by penpornk** [comment]
@gzmkl Please track the cherry-pick progress here: https://github.com/tensorflow/tensorflow/pull/48651 :)

**Comment by gzmkl** [comment]
@penpornk  thank you for let me know the cherry pick process. I will keep an eye on the specified item.

**Pull request merged**